daemons/controld/controld_execd.c

```diff
@@ -632,22 +631,27 @@ append_restart_list(lrmd_event_data_t *op, struct ra_metadata_s *metadata,
     digest = calculate_operation_digest(restart, version);
     /* Add "op-force-restart" and "op-restart-digest" to indicate the resource supports reload,
      * no matter if it actually supports any parameters with unique="1"). */
-    crm_xml_add(update, XML_LRM_ATTR_OP_RESTART, list? list: "");
     crm_xml_add(update, XML_LRM_ATTR_RESTART_DIGEST, digest);
 
-    crm_trace("%s: %s, %s", op->rsc_id, digest, list);
+    if (list != NULL) {
```

Comment:
Currently assumes `list` is non-null. I guess that's probably a safe assumption

Reply:
> We also don't need to trace it twice, either one would be sufficient.

What exactly are you referring to? I'm about to make the regular append mirror the secure append but it sounds like something else is being asked, since they both have the same number of trace calls... Only differences w.r.t. tracing are that `crm_log_xml_trace()` is called unconditionally here and conditionally in secure, and the digest is traced unconditionally here.

Reply:
Looking at `calculate_operation_digest()`, it calls `calculate_xml_digest_v1()`, which traces its input (`restart`)... so maybe that's what you meant. It looks like the same XML will get traced twice here -- granted, tracing might be enabled in only one location.

Reply:
I think I jumped the gun on this change in the first place. It looks like we do still want to do some things even if there are no reloadable or unique params. For now, I'm gonna revert to the original structure and just add NULL checks where applicable.

Reply:
> It looks like we do still want to create a digest and add it to the XML even if list is empty.

Ah right, I overlooked that too. I believe its presence triggers some code.